#!/usr/bin/env python
PACKAGE = "trj_publisher"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("side", bool_t, 0, "True = right", True)
gen.add("publish_frequency", int_t, 0, "publish frequency", 15, 0, 100)
gen.add("wave_period", int_t, 0, "wave period", 4, 0, 10)

gen.add("right_wave_speed", double_t, 0, "right wave speed", 0, 0, 3)
gen.add("right_wave_center_x", double_t, 0, "right wave center x", 0.0, -2, 2)
gen.add("right_wave_center_y", double_t, 0, "right wave center y", 0.0, -2, 2)
gen.add("right_wave_center_z", double_t, 0, "right wave center z", 0.0, -2, 2)
gen.add("right_amplitude_x", double_t, 0, "right amplitude x", 0.00, -0.5, 0.5)
gen.add("right_amplitude_y", double_t, 0, "right amplitude y", 0.00, -0.5, 0.5)
gen.add("right_amplitude_z", double_t, 0, "right amplitude z", 0.00, -0.5, 0.5)

gen.add("left_wave_speed", double_t, 0, "left wave speed", 0, 0, 3)
gen.add("left_wave_center_x", double_t, 0, "left wave center x", 0.0, -2, 2)
gen.add("left_wave_center_y", double_t, 0, "left wave center y", 0.0, -2, 2)
gen.add("left_wave_center_z", double_t, 0, "left wave center z", 0.0, -2, 2)
gen.add("left_amplitude_x", double_t, 0, "left amplitude x", 0.00, -0.5, 0.5)
gen.add("left_amplitude_y", double_t, 0, "left amplitude y", 0.00, -0.5, 0.5)
gen.add("left_amplitude_z", double_t, 0, "left amplitude z", 0.00, -0.5, 0.5)

exit(gen.generate(PACKAGE, "hand_pose_publisher", "TrjPublisher"))
